### DISCUSSION TITLE
Restructure of Website

### DISCUSSION THREAD

**Discussion by toby63** [mumble-voip/mumble-www#132] [General]
We have two colliding problems:
1. I would like to add some more content to the Menu:
- About #128 
- Contact #128 
- Source Code link #129 
- (optional) Donate page

2. The menu is a bit short on space.

There are two solutions:
1. make the menu wider:
    - either by reducing the mumble icon size
    - or by making it wider
2. use drop-down menus (see [here](https://discourse.gohugo.io/t/how-to-create-multilevel-dropdown-menu-in-hugo/18373/5))

---

I think option 1 would be easily possible, because the whole website is already a bit tight and could be made wider in general.
If we use drop-down menus (option 2), the question is, what structure should we use?

One concept:
- Home -> News (now Blog) (either inside or as dropdown, or both)
- Downloads (-> Source Code) - not ideal, but better than nothing.
- Documentation
- Contribute (-> Donate) - does not really fit there I think...
- About (-> Contact) - not ideal, but better than nothing.
- Sponsors (I don't think this point is a must-have in the menu, but I don't know the details)

**Comment by Krzmbrzl** [comment]
Increasing the overall width only solves the issue for wide-screens.

Thus using a dropdown menu is probably the best choice.

We could also leave the menu entries that currently exist as they are (minus the sponsors entry perhaps) and then use a new Entry "More" or "Other" and make that a dropdown that holds all additional entries.

**Reply by toby63** [reply]
> Increasing the overall width only solves the issue for wide-screens.

I don't really see a problem, on mobile devices it will probably jump to the next line even now.
I have not checked that though.

> Entry "More" or "Other" and make that a dropdown that holds all additional entries.

Also an idea :thinking:.
I have to say that I don't like it (as a general concept), but it's probably better than too many entries encapsulated without real relation to each other.

Some suggestions:
- I would name it "more" (and nothing else)
- The most important parts should be directly visible:
     - Home, Downloads, Documentation, About, Contact
     It might be discussable though, which parts are important.

**Reply by Kissaki** [reply]
A typical approach is to present a normal, expanded menu when the space is available, and when screens get smaller, and specifically on mobile, hide the menu behind a dropdown button.

I think we have a few options to structure, lay out and combine menu items and how to present them. And rather than a decision between the two extremes, we have to balance and design some alternatives and work from there. Abstract talk only gets us so far.

Unfortunately I did not find the time yet to look through it and make a suggestion. And the many suggested changes in PRs overwhelmed me, made it more of a barrier, not wanting to cause conflicts too.

**Reply by Krzmbrzl** [reply]
> A typical approach is to present a normal, expanded menu when the space is available, and when screens get smaller, and specifically on mobile, hide the menu behind a dropdown button.

Indeed. I think that'd be nice, too

**Reply by toby63** [reply]
> Abstract talk only gets us so far.

Well, you are free to try different things.
But I think an abstract plan is always a good start.

> And the many suggested changes in PRs overwhelmed me, made it more of a barrier, not wanting to cause conflicts too.

I listed all things above, so I don't really understand the confusion.
Also all of the PRs are almost ready now and won't change much anymore (from my point of you), so they are also ready for review.

> A typical approach is to present a normal, expanded menu when the space is available, and when screens get smaller, and specifically on mobile, hide the menu behind a dropdown button.
> rather than a decision between the two extremes

Just say'in, essentially thats option 1 with a dropdown button for mobile clients.
Still a good idea.
